agent/lib/docker/container.rb

```diff
@@ -216,6 +216,14 @@ def overlay_suffix
       self.overlay_cidr.split('/')[1] if self.overlay_cidr
     end
 
+    # How long to wait when attempting to stop a container if it doesn’t handle
+    # SIGTERM (or whatever stop signal has been specified with stop_signal), before sending SIGKILL
+    #
+    # @return [Integer]
+    def stop_grace_period
+      self.labels['io.kontena.container.stop_grace_period'] ? self.labels['io.kontena.container.stop_grace_period'].to_i : 10
```

Comment:
there's the `to_i` "cast" which would create `0` timeout for all previously existing containers if we used `self.labels[...] || 10`